cpp/main.cpp

```diff
@@ -442,6 +483,118 @@ set_detection_efficiencies(petsird::ScannerInformation& scanner, const ScannerGe
     }
 }
 
+void
+ReadNormalizationFactorFile(petsird::ScannerInformation& scanner, const ScannerGeometry& scannerGeometry, const string& filename)
+{
+  const petsird::TypeOfModule type_of_module{ 0 };
+
+  const auto& rep_module = scanner.scanner_geometry.replicated_modules[type_of_module];
+  //const auto num_modules = rep_module.transforms.size();
+  //printf("num_modules=%ld\n", num_modules);
+  //printf("scannerGeometry.n_rsec_z=%d\n", scannerGeometry.n_rsec_z);
+
+  auto& module_pair_SGID_LUT = (*scanner.detection_efficiencies.module_pair_sgidlut)[type_of_module][type_of_module];
+
+  const auto& detecting_elements = rep_module.object.detecting_elements;
+  const auto num_det_els_in_module = detecting_elements.transforms.size();
+  const auto& event_energy_bin_edges = scanner.event_energy_bin_edges[type_of_module];
+  const auto num_event_energy_bins = event_energy_bin_edges.NumberOfBins();
+  const auto num_detection_bins_in_module = num_det_els_in_module * num_event_energy_bins;
+
+  const unsigned int num_SGIDs = scannerGeometry.n_rsec_z * scannerGeometry.n_rsec_z * (scannerGeometry.n_rsec_xy - 1);
+
+  yardl::NDArray<float, 3> sum_components({num_SGIDs, num_detection_bins_in_module, num_detection_bins_in_module});
+  yardl::NDArray<int, 3> count_components({num_SGIDs, num_detection_bins_in_module, num_detection_bins_in_module});
+
+  //printf("scannerGeometry.n_smod_xy=%d, scannerGeometry.n_smod_z=%d\n", scannerGeometry.n_smod_xy, scannerGeometry.n_smod_z);
+  //printf("scannerGeometry.n_cry_xy=%d, scannerGeometry.n_cry_z=%d, scannerGeometry.n_cry_layers=%d\n", scannerGeometry.n_cry_xy, scannerGeometry.n_cry_z, scannerGeometry.n_cry_layers);
+
+  //getchar();
+
+  std::ifstream fin(filename, std::ios::binary);
+  if (!fin) {
+    std::cerr << "Failed to open normalization file!\n" << std::endl;
+    return;
+  }
+  printf("Reading of the normalization file ..."); fflush(stdout);
```

Comment:
don't mix C-style `printf` and C++-style `std:cout<<`. You're not guaranteed they come out "in order".
```suggestion
  std::cout << "Reading of the normalization file ..." << std::endl;
```